runtime/registry.go

```diff
@@ -79,11 +71,139 @@ func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) er
 	return nil
 }
 
-func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string) error {
-	err := r.Registry().DeleteInstance(ctx, instanceID)
+func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string, dropDB bool) error {
+	inst, err := r.Registry().FindInstance(ctx, instanceID)
 	if err != nil {
+		if errors.Is(err, drivers.ErrNotFound) {
+			return nil
+		}
 		return err
 	}
 
-	return nil
+	svc, err := r.catalogCache.get(ctx, r, instanceID)
+	if err != nil { // return error if db handlers can't be opened
+		return err
+	}
+
+	// delete instance related data if catalog is not embedded
+	if !inst.EmbedCatalog {
+		err := svc.Catalog.DeleteEntries(ctx, instanceID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if dropDB {
+		// ignoring the dropDB error since if db is already dropped it may not be possible to retry
+		// also the only error that os.remove returns is patherror which should be ignored as well
+		_ = svc.Olap.DropDB()
```

Comment:
If there's an error, let's log it (even if not returning) – something like: `r.logger.Error("could not drop database", zap.Error(err), zap.String("instance_id", instanceID))`